components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/utils/CAObjectUtils.java

```
@@ -0,0 +1,143 @@
+import java.io.UnsupportedEncodingException;
+import java.security.PrivateKey;
+import java.security.cert.CRLException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509CRL;
+import java.security.cert.X509Certificate;
+
+/**
+ * Contains the Util methods to encode/decode the certificates, CSRs, CRLs,..
+ */
+public class CAObjectUtils {
+
+    private CAObjectUtils() {
+    }
+
+    /**
+     * Decode base 64 encoded CRL to X509CRL
+     *
+     * @param encodedCrl The crl in base 64 encoded format
+     * @return
+     * @throws org.wso2.carbon.identity.certificateauthority.CAException
+     */
+    public static X509CRL toX509Crl(String encodedCrl) throws CAException {
+        try {
+            byte[] b64DecodeBytes = Base64.decode(encodedCrl.getBytes(CAConstants.UTF_8_CHARSET));
+            CertificateFactory certificateFactory = CertificateFactory.getInstance(CAConstants.X509);
+            return (X509CRL) certificateFactory.generateCRL(new ByteArrayInputStream(b64DecodeBytes));
+        } catch (CRLException e) {
+            throw new CAException("Can't decode CRL.", e);
+        } catch (CertificateException e) {
+            throw new CAException("Couldn't create certificate factory", e);
+        } catch (UnsupportedEncodingException e) {
+            throw new CAException("Error with the charset used", e);
+        }
+    }
+
+    /**
+     * PEM encode a given certificate
+     *
+     * @param certificate The X509 certificate to be encoded
+     * @return
+     */
+    public static String toPemEncodedCertificate(X509Certificate certificate) {
+        try {
+            StringWriter stringWriter = new StringWriter();
+            PEMWriter writer = new PEMWriter(stringWriter);
+            writer.writeObject(certificate);
+            writer.close();
+            return stringWriter.toString();
+        } catch (IOException ignored) {
+            return "";
```

Add a comment stating why you ignore the exception